src/nspawn/nspawn-mount.c

```diff
@@ -815,8 +817,8 @@ static int mount_bind(const char *dest, CustomMount *m, uid_t uid_shift, uid_t u
                         return log_error_errno(r, "Read-only bind mount failed: %m");
         }
 
-        if (idmapped) {
-                r = remount_idmap(where, uid_shift, uid_range, REMOUNT_IDMAP_HOST_ROOT);
+        if (idmapping != REMOUNT_IDMAPPING_NONE) {
+                r = remount_idmap(where, uid_shift, uid_range, source_st.st_uid, idmapping);
```

Comment:
the docs should say somewhere that the owning gid has no effect, we purely look at the uid, and gid is always copied from the uid

Reply:
Thinking about it, shouldn't we map the GID too? For the mkosi use case, for example, it would make sense for files to be owned by my UID and my GID on the host filesystem.